src/Categories/Diagram/End/Fubini.agda

```diff
@@ -0,0 +1,206 @@
+{-# OPTIONS --without-K --lossy-unification --safe #-}
+
+open import Level
+open import Data.Product using (Σ; _,_; _×_) renaming (proj₁ to fst; proj₂ to snd)
+open import Function using (_$_)
+
+open import Categories.Category
+open import Categories.Category.Construction.Functors
+open import Categories.Category.Product renaming (Product to _×ᶜ_)
+open import Categories.Diagram.End using () renaming (End to ∫)
+open import Categories.Diagram.End.Properties
+open import Categories.Diagram.End.Parameterized renaming (EndF to ⨏)
+open import Categories.Diagram.Wedge
+open import Categories.Functor renaming (id to idF)
+open import Categories.Functor.Bifunctor
+open import Categories.Functor.Bifunctor.Properties
+open import Categories.NaturalTransformation using (NaturalTransformation)
+open import Categories.NaturalTransformation.Dinatural
+
+import Categories.Morphism as M
+import Categories.Morphism.Reasoning as MR
+
+module Categories.Diagram.End.Fubini where
+variable
+  o ℓ e : Level
+  C P D : Category o ℓ e
+
+module Functor-Swaps  (F : Bifunctor (Category.op C ×ᶜ Category.op P) (C ×ᶜ P) D) where
+  private
+    module C = Category C
+    module P = Category P
+
+  -- just shuffling arguments
+  -- An end of F ′ can be taken iterated, instead of in the product category
+  _′ : Bifunctor (P.op ×ᶜ P) (C.op ×ᶜ C) D
+  _′ = F ∘F ((πˡ ∘F πʳ ※ πˡ ∘F πˡ) ※ (πʳ ∘F πʳ ※ πʳ ∘F πˡ))
+
+  -- An end of F ′′ is the same as F, but the order in the product category is reversed
+  _′′ : Bifunctor (Category.op P ×ᶜ Category.op C) (P ×ᶜ C) D
+  _′′ = F ∘F (Swap ⁂ Swap)
+
+open Functor-Swaps
+
+module _  (F : Bifunctor (Category.op C ×ᶜ Category.op P) (C ×ᶜ P) D)
+          {ω : ∀ ((p , q) : Category.Obj P × Category.Obj P) → ∫ (appˡ (F ′) (p , q))}
+           where
+  open M D
+  private
+    module C = Category C
+    module P = Category P
+    module D = Category D
+    open module F = Functor F using (F₀; F₁; F-resp-≈)
+    ∫F = (⨏ (F ′) {ω})
+    module ∫F = Functor ∫F
+
+    module ω ((p , q) : P.Obj × P.Obj) = ∫ (ω (p , q))
+
+  open _≅_
+  open Iso
+  open DinaturalTransformation using (α)
+  open NaturalTransformation using (η)
+  open Wedge renaming (E to apex)
+  open Category D
+  open import Categories.Morphism.Reasoning D
+  open HomReasoning
+  -- we show that wedges of (∫.E e₀) are in bijection with wedges of F
+  -- following MacLane §IX.8
+  private module _ (ρ : Wedge ∫F) where
+    private
+      open module ρ = Wedge ρ using () renaming (E to x)
+    -- first, our wedge gives us a wedge in the product
+    ξ : Wedge F
+    ξ .apex = x
+    ξ .dinatural = dtHelper record
+      { α = λ { (c , p) → ω.dinatural.α (p , p) c ∘ ρ.dinatural.α p }
+      ; commute = λ { {c , p} {d , q} (f , s) → begin
```

Comment:
Please pull out this inline proof of `commute` into a `where` clause below the record. You can then take advantage of this to declare some short-forms for things that are repeated a lot (and long) like `ω.dinatural.α` and `ρ.dinatural.α p` and likely more.  Even just naming `(C.id , P.id)` would save a lot of horizontal space.